packages/react-native-codegen/src/parsers/parser.js

```diff
@@ -39,4 +39,10 @@ export interface Parser {
    * @returns: the name of the type.
    */
   nameForGenericTypeAnnotation(typeAnnotation: $FlowFixMe): string;
+  /**
+   * Given the content of a file and options, it returns an AST.
+   * @parameter contents: the content of the file.
+   * @returns: the AST of the file (given in program property for typescript).
+   */
+  parse(contents: string): $FlowFixMe | {program: $FlowFixMe};
```

Comment:
I think that the return types should be `SchemaType`. This should also help you removing various comments for the linter.